lib/mix/tasks/up_ex_doc_version.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Mix.Tasks.UpExDocVersion do
+  use Mix.Task
+
+  @shortdoc "Update mix.exs' ex_doc version"
+  @mix_exs "mix.exs"
+
+  def run([new_vsn]) do
+    content_up =
+      @mix_exs
+      |> File.read!()
+      |> run(new_vsn)
+
+    File.write!(@mix_exs, "#{content_up}\n")
+    Mix.Task.run("deps.unlock", ["ex_doc"])
+    Mix.Task.run("deps.get")
```

Comment:
```suggestion
    env = [{"PATH", System.get_env("PATH")}]
    System.cmd("mix", ["deps.unlock", "ex_doc"], env: env)
    System.cmd("mix", ["deps.get"], env: env)
```

Reply:
I had an off-by-one issue here. Mix's state is not updated when we write the file, so `Mix.Task.run` would only work on the second update.